src/Microsoft.Data.Sqlite/SqliteDataReader.cs

```diff
@@ -94,7 +98,11 @@ public override bool Read()
                 return _hasRows;
             }
 
-            var rc = NativeMethods.sqlite3_step(_stmt);
+            var rc = NativeMethods.sqlite3_step_blocking(Command.Connection.DbHandle, _stmt, Command.CommandTimeout*1000);
```

Comment:
Good point. I hadn't tested with just the first call being blocking. I'll test for it.